educator/src/Dscp/DB/SQLite/Queries.hs

```diff
@@ -422,9 +424,15 @@ createBlock delta = do
 
         hdr = PrivateBlockHeader prev root trueDelta
 
+    let isNullBlock = and
+            [ nullATGDelta trueDelta
+            , null txs
+            ]
+    guard (not isNullBlock)
```

Comment:
Why don't you just throw an exception out of this function?
Later the output is just mostly ignored.
This is an inconsistency of this function behaviour, which can mess up things later.

I'm proposing throwing `NoNeedToCreateANewBlock` and `catch`ing it on the call sites instead, to show that this thing might happen.

Reply:
For me semantics of `m (Maybe a)` is "function performs an action and returns or does not return something", while `MaybeT m a` means "action may be performed or not" which is exactly what we want here.